gap/Parser.gi

```diff
@@ -238,6 +246,19 @@ InstallGlobalFunction( AutoDoc_Parser_ReadFiles,
             current_line := StripBeginEnd( current_line, " " );
             current_item!.name := current_line{ [ 1 .. Minimum( [ PositionSublist( current_line, "," ), PositionSublist( current_line, ");" ) ] ) - 1 ] };
             current_item!.name := StripBeginEnd( ReplacedString( current_item!.name, "\"", "" ), " " );
+            if IsBound(current_item!.coll_suffix) then
+                if EndsWith(current_item!.name, "Collection") then
+                    current_item!.name :=
+                    current_item!.name{[1..Length(current_item!.name)-6]};
+                fi;
+                if EndsWith(current_item!.name, "Coll") then
+                    current_item!.coll_suffix := "Coll";
+                else
+                    current_item!.coll_suffix := "Collection";
+                fi;
+                current_item!.name := Concatenation(current_item!.name,
+                                                    current_item!.coll_suffix);
+            fi;
```

Comment:
Can you please explain what this `if` does?

To me it seems like you check if the item ends with `Collection`, you do nothing. If it ends with `Coll`, you add another `Coll` to the name?

Specially, in the docs below it seems like `MyThingsColl` gets a manual entry `MyThingsCollColl`, which seems wrong . . .

Reply:
As explained in the PR description and the commit message, this imitates the behavior of `DeclareCategoryCollections`. I recommend looking at the code for `DeclareCategoryCollections` (and the comment before it) in the GAP library. Let me know if there are still questions after that :-).

That said, I think there should be a comment here which explains that; I can add one later.

Reply:
I see . . .

Okay, sorry for the confusion.

Reply:
No confusion, nothing to be sorry about: you raised a valid point. I've now added a comment, which hopefully makes this code easier to understand for future generations (e.g. us, two years down the road ;-).

Reply:
I will also submit a PR to GAP to actually document `DeclareCategoryCollections`, which lots of packages use...

In the meantime, feel free to approve this PR ;-)